libcontainer/notify_linux_v2.go

```diff
@@ -0,0 +1,92 @@
+// +build linux
+
+package libcontainer
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/fsnotify/fsnotify"
```

Comment:
This package is not well maintained, and tries to work for all systems, sacrificing linux-specific features.

I havent't checked it, but can you take a look at how hard it would be to use `Inotify*` functions from `golang.org/x/sys/unix` directly?